### PR TITLE
235 new comment textarea

### PR DESCRIPTION
**Pull request by yishain11**

### CODE REVIEW

**Review by avirankatz — COMMENTED** [review]
- [x] Revert core_comments.yaml

**Review by avirankatz — CHANGES_REQUESTED** [review]
- [x] Revert core_comments.yaml
- [ ] Revert core_section_versions.yaml
- [ ] Rename AddComment (constant in mutations) to addComment
- [ ] Rename addCommentVars (interface in section context) to AddCommentVars
- [ ] authorId inside add comment vars should be automatically inserted in the db (trigger), we shouldn't send it as a variable
- [ ] Replace ContentEditor with a simple TextArea, we don't allow styling in comments
- [ ] Submit comment button > onclick: extract handler to component scope
- [ ] Section.tsx:264 > use 'Publish' from string bank

**Review by avirankatz — CHANGES_REQUESTED** [review]
- [ ] TextAreaAutosize - extract on change handler to component scope
- [ ] remove author from add comment vars
- [ ] style: avoid hard coded widths and heights (Section.tsx box and text area)
- [ ] style: use color const instead of hard coded string (Section.tsx:253)
- [ ] text area: verify that all styling is necessary; remove unnecessary styling
- [ ] text area: remove ref and usages
- [ ] Section.tsx:271 - remove hard coded color string; use const from theme instead

File: frontend/src/pages/Section.tsx [hunk 1]
```diff
@@ -190,13 +214,85 @@ const Section: FC = () => {
...
+                  onClick={handelAddComment}
+                >
+                  <Typography
+                    color={
+                      newComment === '' || txtAreaEl.current?.value === '' ? 'gray' : '#c49eff'
+                    }
+                  >
+                    {t(StringBank.PUBLISH)}
+                  </Typography>
+                </Button>
```
Comment (avirankatz) [inline frontend/src/pages/Section.tsx:276]:
```suggestion
                <TextField
                  placeholder={t(StringBank.ADD_COMMENT_IN_SECTION)}
                  onChange={(data) => {
                    const commentText = data.target.value;
                    if (commentText !== '') {
                      setNewComment(commentText);
                      return;
                    }
                    data.target.value = '';
                    setNewComment('');
                  }}
                  minRows={3}
                  multiline
                  fullWidth
                />
              </Stack>
              <Stack direction="row" justifyContent="end">
                <Button
                  disabled={!newComment}
                  sx={{ paddingX: 0, marginY: 1 }}
                  onClick={handelAddComment}
                >
                  {t(StringBank.PUBLISH)}
                </Button>
```

**Review by avirankatz — COMMENTED** [review]
See suggestion for text area

**Comment by yishain11** [comment]
> * [x]  TextAreaAutosize - extract on change handler to component scope
> * [x]  remove author from add comment vars
> * [ ]  style: avoid hard coded widths and heights (Section.tsx box and text area)
> * [ ]  style: use color const instead of hard coded string (Section.tsx:253)
> * [ ]  text area: verify that all styling is necessary; remove unnecessary styling
> * [ ]  text area: remove ref and usages
> * [x]  Section.tsx:271 - remove hard coded color string; use const from theme instead

text area: remove ref and usages: 
this is needed in order to make the comment box empty after publishing.

**Pull request merged**